### DISCUSSION TITLE
Rename the 'tools' directory to 'tests'

### DISCUSSION THREAD

**Discussion by JohnVillalovos** [python-gitlab/python-gitlab#1468] [General]
To me it would make sense to rename the `tools/` directory to `tests/`.

At the moment there only appear to be the functional tests underneath the `tools/` directory.

And then after that we could create:  `tests/unit/` in addition to `tests/functional/`. Idea being to move `gitlab/tests/` to `tests/unit/`

Consolidate all the tests under the one directory `tests/`

Thoughts?

**Comment by nejch** [comment]
Yes absolutely! :) That was also my intention a while ago, see https://github.com/python-gitlab/python-gitlab/pull/1024 that was never finished (this was pre-pytest even).

Just need to make sure that `tests/unit` is packaged along with it, as that's how some package maintainers test it (e.g. for rpm/deb distros), but probably not `tests/functional` as that has a lot more system dependencies.

However there's some quirks with pytest that might make a `src/gitlab` layout a good idea, but I see now there's also an import fix for that, see https://docs.pytest.org/en/6.2.x/goodpractices.html?highlight=directory#choosing-a-test-layout-import-rules

**Comment by JohnVillalovos** [comment]
The `tools/` directory is now named `tests/` and the functional tests are now there. [1]

Next step (at some point) will be to move the `gitlab/tests/` directory to `tests/unit/`


[1] Commit: https://github.com/python-gitlab/python-gitlab/pull/1469
